.github/workflows/solidity-foundry.yml

```diff
@@ -11,28 +11,55 @@ jobs:
     name: Detect changes
     runs-on: ubuntu-latest
     outputs:
-      changes: ${{ steps.changes.outputs.src }}
+      src_changes: ${{ steps.changes.outputs.src }}
```

Comment:
I think it makes sense to check for changes in the folders per product. Right now we would be running coverage for all products if any single one changes a line. Ideally, CI only runs for affected products. 

Any changes to `vendor` or `shared` would trigger a full run, don't think we can get around that